packages/clay-dropdown/src/ClayDropdownBase.js

```diff
@@ -121,6 +140,165 @@ class ClayDropdownBase extends ClayComponent {
+
+				subActive = items[active].length - 1;
+			}
+		} else {
+			if (subActive === items[active].length) {
+				const nextActive = active + 1;
+
+				active = nextActive === totalItems ? 0 : nextActive;
+
+				subActive = 0;
+			}
+		}
+
+		return {active, subActive};
+	}
+
+	/**
+	 * Gets the indexes of the next active item.
+	 * @param {Array} items
+	 * @param {Number} initialActive
+	 * @param {Number} initialSubActive
+	 * @param {Boolean} reverse
+	 * @private
+	 * @return {Number}
+	 */
+	_getNextIndexes(items, initialActive, initialSubActive, reverse = false) {
+		let {active, subActive} = this._incrementActiveItemIndex(
+			items,
+			initialActive === -1 ? 0 : initialActive,
+			initialSubActive,
+			reverse
+		);
+
+		const totalItems = items.flat(2);
+
+		let tick = 0;
+
+		while (items[active][subActive].disabled && tick < totalItems.length) {
+			const item = this._incrementActiveItemIndex(
+				items,
+				active,
+				subActive,
+				reverse
+			);
+
+			active = item.active;
+			subActive = item.subActive;
+
+			tick++;
+		}
+
+		return {active, subActive};
+	}
+
+	/**
+	 * Sets the next item in the list as active.
+	 * @param {Boolean} reverse
+	 * @private
+	 */
+	_setNextActive(reverse) {
+		const items = flatten(this.items, true);
+		let activeSubIndex = -1;
+
+		const activeIndex = items.findIndex(item => {
+			const subIndex = item.findIndex(item => item.active);
+
+			if (subIndex !== -1) {
+				activeSubIndex = subIndex;
+
+				return true;
+			}
+		});
+
+		const {active, subActive} = this._getNextIndexes(
+			items,
+			activeIndex,
+			activeSubIndex,
+			reverse
+		);
+
+		if (items[activeIndex] && items[activeIndex][activeSubIndex]) {
+			items[activeIndex][activeSubIndex].active = false;
+		}
+
+		items[active][subActive].active = true;
+
+		this.items = this._assembleFromFlattenedGroups(items);
+	}
+
+	/**
+	 * Reassembles flattened items into the correct items structure.
+	 * This method is used in conjunction with `flatten(..., true)`
+	 * @param {Array} flattenedItems
+	 * @private
+	 * @return {Array}
+	 */
+	_assembleFromFlattenedGroups(flattenedItems) {
+		return this.items.map((item, i) => {
+			if (item.items) {
+				item.items = flattenedItems[i];
+			} else {
+				item = flattenedItems[i][0];
+			}
+
+			return item;
+		});
+	}
+
+	/**
+	 * Handle key events.
+	 * @param {!Event} event
+	 * @private
+	 */
+	_handleKeyNavigation(event) {
+		if (KEY_NAVIGATION.includes(event.key)) {
+			event.preventDefault();
+
+			if (event.key === KEY_DOWN || event.key === KEY_UP) {
+				this._setNextActive(event.key === KEY_UP);
+			} else if (event.key === KEY_ENTER) {
+				this._handleEnterKey(event);
+			}
+		}
+	}
+
+	/**
+	 * Continues the propagation of the enter key event
+	 * @param {!Event} event
+	 * @protected
+	 */
+	_handleEnterKey(event) {
+		this.emit({
+			data: {
+				item: flatten(this.items).find(item => item.active),
+			},
+			name: 'itemSelected',
```

Comment:
hey @bryceosterhaus, can you propagate this event to the other components as well?

- ClayActionsDropdown
- ClayCreationMenuDropdown
- ClayDropdown

Reply:
hey @bryceosterhaus, Oh yeah, ClayDropdown follows Lexicon standards as stated in this [document](https://docs.google.com/document/d/1eZkT4XZuA7KfxcgQUwdMdlJxfeUeKNtJ_gjFSQdPRTg/edit#heading=h.owd64wgehjjl), so we have some Dropdown collections here, in the cases above. That way we ensure that people follow the correct Lexicon standards.

We have ClayDropdownBase to be used as a more flexible and somewhat generic component that helps us to reuse it in other parts of Clay components if necessary. We really do not recommend that people use ClayDropdownBase (Maybe use to create other cases...), but rather these others.

We call ClayDropdownBase in the above variations, restricting some APIs and especially the `label` API.

Hope I have been clear 😅, please let me know if you have any further questions about this.

Reply:
Oh yeah that makes sense. I just pushed up another commit to propagate the event